### PR TITLE
Ikg tests

### PR DESCRIPTION
**Pull request by cowan-macady**
- add ikg tests
- update function GetNodeByIdentifier (modification in type management)
- change examples

### CODE REVIEW

File: test/constants.go [hunk 1]
```diff
@@ -275,6 +279,79 @@ var (
...
+			"Id":         gomega.Equal("gid:AAAAHO-ocNjhAU1dlkF_1QG22Uo"),
+			"ExternalId": gomega.Equal("fVcaUxJqmOkyOTX"),
+			"Type":       gomega.Equal("Person"),
+		}))),
+		"Relationships": gomega.BeEmpty(),
+	}
+	Matcher2 = gstruct.Fields{
+		"Nodes": gomega.ContainElement(gstruct.PointTo(gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+			"Id":         gomega.Equal("gid:AAAAHMtQUEbcWkgHkjAQPM9MOyo"),
+			"ExternalId": gomega.Equal("Truck1"),
```
Comment (jakubtomany) [inline test/constants.go:340]:
```suggestion
			"Id":         gomega.Equal(Truck1Id),
			"ExternalId": gomega.Equal(Truck1),
```

File: test/constants.go [hunk 2]
```diff
@@ -275,6 +279,79 @@ var (
...
+			"Type":       gomega.Equal("Truck"),
+		}))),
+		"Relationships": gomega.BeEmpty(),
+	}
+	Matcher3 = gstruct.Fields{
+		"Id":         gomega.Equal("gid:AAAAHMtQUEbcWkgHkjAQPM9MOyo"),
+		"ExternalId": gomega.Equal("Truck1"),
```
Comment (jakubtomany) [inline test/constants.go:347]:
```suggestion
			"Id":         gomega.Equal(Truck1Id),
			"ExternalId": gomega.Equal(Truck1),
```

File: test/constants.go [hunk 3]
```diff
@@ -275,6 +279,79 @@ var (
...
+	}
+	Returns1 = []*knowledgepb.Return{
+		{
+			Variable: "n",
+		},
+	}
+	Matcher1 = gstruct.Fields{
+		"Nodes": gomega.ContainElement(gstruct.PointTo(gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+			"Id":         gomega.Equal("gid:AAAAHO-ocNjhAU1dlkF_1QG22Uo"),
+			"ExternalId": gomega.Equal("fVcaUxJqmOkyOTX"),
```
Comment (jakubtomany) [inline test/constants.go:332]:
```suggestion
			"Id":         gomega.Equal(Node2),
			"ExternalId": gomega.Equal(Subject2),
```

File: test/constants.go [hunk 4]
```diff
@@ -275,6 +279,79 @@ var (
...
+	Query2 = `MATCH (n:Person)-[:BELONGS_TO]->(o:Organization)-[:OWNS]->(t:Truck)-[:HAS]->(p:Property:External)
+	 WHERE n.external_id=$external_id AND n.type=$type`
+	Query3  = "MATCH (n:Person)-[:BELONGS_TO]->(o:Organization)-[:OWNS]->(t:Truck)"
+	Query4  = "MATCH (n:Person)-[:BELONGS_TO]->(o:Organization)-[:OWNS]->(t:Truck)-[:HAS]->(p:Property:External)"
+	Query5  = "MATCH (n:Resource)-[:HAS]->(p:Property) WHERE p.type=$type and p.value=$value"
+	Query6  = "MATCH (n:Resource)-[:HAS]->(p:Property:External) WHERE p.type=$type and p.value=$value"
+	Query7  = "Person "
+	Params1 = map[string]*objects.Value{
+		"external_id": {
+			Type: &objects.Value_StringValue{StringValue: "fVcaUxJqmOkyOTX"},
```
Comment (jakubtomany) [inline test/constants.go:293]:
```suggestion
			Type: &objects.Value_StringValue{StringValue: Node2},
```

**Review by jakubtomany — CHANGES_REQUESTED** [review]
Use the existing constants or create new constants as the rest of the code here.

File: test/constants.go (hunk 1 above)
Comment (cowan-macady) [reply test/constants.go]: updated

File: test/constants.go (hunk 3 above)
Comment (cowan-macady) [reply test/constants.go]: updated

File: test/constants.go (hunk 4 above)
Comment (cowan-macady) [reply test/constants.go]: updated

File: test/constants.go (hunk 2 above)
Comment (cowan-macady) [reply test/constants.go]: updated

**Review by jakubtomany — APPROVED** [review]

**Pull request merged**